pg-index-health-model/src/main/java/io/github/mfvanek/pg/model/sequence/SequenceState.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
+ * https://github.com/mfvanek/pg-index-health
+ *
+ * This file is a part of "pg-index-health" - a Java library for
+ * analyzing and maintaining indexes health in PostgreSQL databases.
+ *
+ * Licensed under the Apache License 2.0
+ */
+
+package io.github.mfvanek.pg.model.sequence;
+
+import io.github.mfvanek.pg.model.DbObject;
+import io.github.mfvanek.pg.model.validation.Validators;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A representation of a sequence's state in a database.
+ * This class encapsulates the details of a database sequence, including its name, data type,
+ * and the percentage of remaining values before it overflows.
+ *
+ * @since 0.11.0
+ */
+@Immutable
+public class SequenceState implements DbObject {
+
+    private final String sequenceName;
+    private final String dataType;
+    private final double remainingPercentage;
+
+    /**
+     * Constructs a {@code SequenceState} object.
+     *
+     * @param sequenceName       sequence name; should be non-blank.
+     * @param dataType           data type; should be non-blank.
+     * @param remainingPercentage remaining percentage; should be non-null.
+     */
+    protected SequenceState(
+        @Nonnull final String sequenceName,
+        @Nonnull final String dataType,
+        final double remainingPercentage
+    ) {
+        this.sequenceName = Validators.notBlank(sequenceName, "sequenceName");
+        this.dataType = Validators.notBlank(dataType, "dataType");
+        this.remainingPercentage = remainingPercentage;
```

Comment:
Should be positive and less than or equal to 100. Take a look at `validPercent` method.